### DISCUSSION TITLE
How to template table data in an email notification

### DISCUSSION THREAD

**Discussion by celexus** [solspace/craft-freeform#441] [Questions & Answers] [answered]
I'm using "Manual Email Body" for my notification templates
It's nice and straightforward to output values from textfields like {{ firstName }}, but I can't seem to figure out how to output saved table data. 
Can anyone please point me in the right direction ?

**Comment by kjmartens** [comment, marked as answer]
This is a fairly complex field type, but it can/should be done in a somewhat automated fashion:
https://docs.solspace.com/craft/freeform/v4/overview/fields.html#submission-data

You'll want to make sure you access the `submission` object in your email notification template for this to work:
https://docs.solspace.com/craft/freeform/v4/overview/email-notifications.html#using-submission-object

```
{% for field in submission %}
    <td>
	{% if field.type == "table" %}
		<table class="table">
			<thead>
				<tr>
				{% for column in field.tableLayout %}
					<th>{{ column.label|default("-") }}</th>
				{% endfor %}
				</tr>
			</thead>
			<tbody>
			{% for row in submission[field.handle].value %}
				<tr>
				{% for index, column in field.tableLayout %}
					<td>{{ row[index] }}</td>
				{% endfor %}
				</tr>
			{% endfor %}
			</tbody>
		</table>
	{% else %}
		{{ submission[field.handle] }}
	{% endif %}
    </td>
{% endfor %}
```

**Reply by celexus** [reply]
Thank you so much this works a charm !
